protocol/avdl/device.avdl

```diff
@@ -7,6 +7,21 @@ protocol device {
     List devices for the user.
     */
   array<Device> deviceList(int sessionID);
+ 
+  record DeviceDetail {
+    Device device;
+    boolean eldest;
+    union {null, Device} provisioner;
```

Comment:
I think adding `Time lastActivity` and `string clientVersion` when we can provide that information seems much simpler than an array where a string determines the type instead of the field name.  

The consumer of this would need to switch on strings instead of using field names, which seems more error-prone than letting the field names be part of the protocol.